### DISCUSSION TITLE
PyQtgraph with PyQt6

### DISCUSSION THREAD

**Discussion by kernd** [pyqtgraph/pyqtgraph#1924] [Q&A] [answered]
Hi,

First off, sorry if this question is already answered, but I have sent some time googling and searching GitHub.

I am trying to run the pyqtgraph examples with PyQt6.

I have the following installed:

```
pip freeze
altgraph==0.17
astroid==2.5.8
autopep8==1.5.5
boto3==1.17.92
botocore==1.20.105
certifi==2021.5.30
cffi==1.14.5
chardet==4.0.0
cryptography==3.4.6
cycler==0.10.0
docopt==0.6.2
getch==1.0
idna==2.10
isort==5.9.1
jmespath==0.10.0
kiwisolver==1.3.1
lazy-object-proxy==1.6.0
macholib==1.14
matplotlib==3.4.1
mccabe==0.6.1
mdict==1.0.3
numpy==1.21.0
pandas==1.2.4
Pillow==8.3.0
playsound==1.2.2
pycodestyle==2.7.0
pycparser==2.20
pyinstaller==4.2
pyinstaller-hooks-contrib==2021.2
pylint==2.7.2
pyOpenSSL==20.0.1
pyparsing==2.4.7
PyQt6==6.0.3
PyQt6-Qt6==6.1.1
PyQt6-sip==13.1.0
pyqtgraph @ git+https://github.com/pyqtgraph/pyqtgraph@7009084e4c0bda7a059964c3484ec1fcf2836950
pyserial==3.5
python-dateutil==2.8.1
pytz==2021.1
regex==2020.11.13
requests==2.25.1
s3transfer==0.4.2
six==1.16.0
toml==0.10.2
urllib3==1.26.6
wrapt==1.12.1
```

I get an error running the examples:

`python -m pyqtgraph.examples
Traceback (most recent call last):
  File "/Users/danny/.pyenv/versions/3.9.4/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/danny/.pyenv/versions/3.9.4/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/Users/danny/.pyenv/versions/ss-Samsara_Tools-3.9/lib/python3.9/site-packages/pyqtgraph/examples/__main__.py", line 16, in <module>
    run()
  File "/Users/danny/.pyenv/versions/ss-Samsara_Tools-3.9/lib/python3.9/site-packages/pyqtgraph/examples/ExampleApp.py", line 540, in main
    loader = ExampleLoader()
  File "/Users/danny/.pyenv/versions/ss-Samsara_Tools-3.9/lib/python3.9/site-packages/pyqtgraph/examples/ExampleApp.py", line 306, in __init__
    self.ui.setupUi(self.cw)
  File "/Users/danny/.pyenv/versions/ss-Samsara_Tools-3.9/lib/python3.9/site-packages/pyqtgraph/examples/exampleLoaderTemplate_pyqt6.py", line 19, in setupUi
    self.splitter.setOrientation(QtCore.Qt.Orientation.Horizontal)
AttributeError: type object 'Qt' has no attribute 'Orientation'`

Does anyone know what is the incompatibility here?

I am running macOS Catalina 10.15.7 and python 3.9.4

Thanks.

**Comment by j9ac9k** [comment, marked as answer]
Hi @kernd 

We don't support Qt 6.0.x, we right now support Qt 6.1+; looks like you have Qt 6.0.3 installed, I would try upgrading that first.

**Reply by kernd** [reply]
HI @j9ac9k , thanks for the quick response. I read the latest release only supports 6.1 in the documentation and for some reason I glossed over my installed version. I think I was looking at PyQt6-Qt6 instead of PtQt6.

I ran `pip uninstall PyQt6` and `pip install PyQt6` and it works now! Thanks!

**Reply by j9ac9k** [reply]
Glad to hear it.  FWIW Qt 6.0.x versions had a lot of changes between patch versions, and the code complexity to support all the released versions was just too much.  We didn't see many users being impacted (who would go to bleeding edge Qt 6 and not go to Qt 6.1 once it was released?)

**Reply by j9ac9k** [reply]
if you don't mind marking the question as answered, that would be appreciated 👍🏻
